hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Path.java

```diff
@@ -196,8 +196,7 @@ public Path(String pathString) throws IllegalArgumentException {
     // parse uri scheme, if any
     int colon = pathString.indexOf(':');
     int slash = pathString.indexOf('/');
-    if ((colon != -1) &&
-        ((slash == -1) || (colon < slash))) {     // has a scheme
+    if ((colon != -1) && (colon == (slash-1))) {     // has a scheme
```

Comment:
@vrozov You have a point - if I change behavior here to colon > 0 it will support parts of path that are just ":".
The only thing is different behavior when you merge two paths - `URI.resolve()` resolves it differently
E.g:
`/abc/:` and `:/:/` versus `/abc/:` and `/:/:/` 
In first case result will be `/abc/:/:/:/` and in the second case `/abc/:/:/`

Reply:
@yufeldman In the first case, you merge absolute and relative path, in the second case, there is an attempt to merge two absolute path, should they be handled differently if merging two absolute path should be allowed at all?